app.js

```diff
@@ -13,22 +13,31 @@ var config = {
 
 require('./index.js').connect(function (context) {
 
+    context.fs.readdir(context.path.join(__dirname, 'api', 'models'), (err, files) => {
+        //Load ALL modules.
+        //Don't get confused by weird logging order! console.log is async and will mix up the actual loading order.
+        files.forEach(file => {
```

Comment:
Actually, forEach is synchronous in JavaScript. 
See [JavaScript, Node.js: is Array.forEach asynchronous?](http://stackoverflow.com/questions/5050265/javascript-node-js-is-array-foreach-asynchronous)

Your example describes the case where an asynchronous function is explicitly called within the forEach loop, in that case it would be asynchronous.